pkg/engine/mutate/strategicPreprocessing_test.go

```diff
@@ -125,6 +125,47 @@ func Test_preProcessStrategicMergePatch_AnnotationMap(t *testing.T) {
 	}
 }
 
+func Test_preProcessStrategicMergePatch_DifferentKinds(t *testing.T) {
+	rawPolicy := []byte(`{
+		"spec": {
+		  "imagePullSecrets": [
+			{
+			  "name": "regcred"
+			}
+		  ],
+		  "initContainers": [
+			{
+			  "(image)": "index.docker.io/*"
+			}
+		  ]
+		}
+	  }`)
+
+	rawResource := []byte(`{
```

Comment:
Can we add these tests:
- resource with `initContainers` that matches the image
- resource with `initContainers` that does not match the image